File: src/hema/schemas/calendar.py
"""Pydantic schemas for Calendar data structures."""

from datetime import date

from pydantic import BaseModel

from hema.schemas.events import EventResponse


class CalendarDay(BaseModel):
    """Single day in the calendar grid."""

    date: date
    is_today: bool
    is_current_month: bool
    events: list[EventResponse]


class CalendarMonth(BaseModel):
    """Complete calendar month data for template rendering."""

    date: date
    days: list[CalendarDay]
    prev_date: date
    next_date: date

File: frontend/types/calendar.ts
import { Event } from './event';

export interface CalendarDay {
  date: string; // YYYY-MM-DD
  is_today: boolean;
  is_current_month: boolean;
  events: Event[];
}

export interface CalendarMonth {
  date: string;
  days: CalendarDay[];
  prev_date: string;
  next_date: string;
}
